framework/packages/abstract-client/src/client.rs

```diff
@@ -119,3 +120,56 @@ impl<Chain: MutCwEnv> AbstractClient<Chain> {
             .map_err(Into::into)
     }
 }
+
+pub mod daemon {
+    use abstract_core::{objects::AccountId, MANAGER};
+    use abstract_interface::AbstractAccount;
+    use cw_orch::{
+        daemon::Daemon,
+        environment::TxHandler as _,
+        state::{ChainState, StateInterface},
+    };
+
+    use super::*;
+
+    impl AbstractClient<Daemon> {
+        /// Retrieve the last account created by the client.
+        /// Returns `None` if no account has been created yet.
+        pub fn get_last_account(&self) -> AbstractClientResult<Option<Account<Daemon>>> {
+            let addresses = self.environment().state().get_all_addresses()?;
+
+            // Now search for all the keys that start with "abstract:manager-x" and return the one which has the highest x.
+            let mut last_account: Option<(u32, Account<Daemon>)> = None;
+            for (id, _) in addresses {
+                if !id.starts_with(MANAGER) {
+                    continue;
+                }
+
+                let account_id_str = id.rsplitn(2, '-').next().unwrap();
```

Comment:
Wouldn't it take only sequence here, for example
```
"abstract:manager-local-5".rsplitn(2, '-').next().unwrap() == "5" 
```
I would prefer to use `split_once` here

Reply:
Ah yeah you're right!